libraries/ESP8266WiFiMesh/src/ESP8266WiFiMesh.cpp

```
@@ -18,151 +18,582 @@
+
+ESP8266WiFiMesh::ESP8266WiFiMesh(std::function<String(String, ESP8266WiFiMesh *)> requestHandler, std::function<transmission_status_t(String, ESP8266WiFiMesh *)> responseHandler, 
+                                 std::function<void(int, ESP8266WiFiMesh *)> networkFilter, String mesh_password, String mesh_name, String node_id, bool verbose_mode, uint8 mesh_wifi_channel, int server_port) 
+                                 : _server(server_port), _lwip_version{0, 0, 0}
+{
+  updateNetworkNames(mesh_name, (node_id != "" ? node_id : Uint64ToString(ESP.getChipId())));
+  _requestHandler = requestHandler;
+  _responseHandler = responseHandler;
+  _mesh_wifi_channel = mesh_wifi_channel;
+  _server_port = server_port;
+  _mesh_password = mesh_password;
+  _verbose_mode = verbose_mode;
+  _networkFilter = networkFilter;
+  storeLwipVersion();
+}
+
+void ESP8266WiFiMesh::verboseModePrint(String string_to_print, bool newline)
+{
+  if(_verbose_mode)
+  {
+    if(newline)
+      Serial.println(string_to_print);
+    else
+      Serial.print(string_to_print);
+  }
+}
+
+/**
+ * Note that using a base higher than 16 increases likelihood of randomly generating ssid strings containing controversial words. 
+ * 
+ * @param number The number to convert to a string with radix "base".
+ * @param base The radix to convert "number" into. Must be between 2 and 36.
+ * @returns A string of "number" encoded in radix "base".
+ */
+String ESP8266WiFiMesh::Uint64ToString(uint64_t number, byte base)
+{
+  String result = "";
+
+  while(number > 0)
+  {
+    result = String((uint32_t)(number % base), base) + result;
+    number /= base;
+  }
+
+  return (result == "" ? "0" : result);
+}
+
+/**
+ * Note that using a base higher than 16 increases likelihood of randomly generating ssid strings containing controversial words. 
+ * 
+ * @param string The string to convert to uint64_t. String must use radix "base".
+ * @param base The radix of "string". Must be between 2 and 36.
```

Is this 2-36 range enforced somehow?

---

No. The problem is that I’m not sure how I would do that in a nice way since I can’t throw exceptions in this environment (as far as I know). The core conversion functions used by my code (`String()` and `strtoul()`) also seem to accept arguments that are out of range, so I just propagated that behaviour through my functions. Of course I could clamp the values to the 2-36 range, but then I would silently convert invalid input arguments to other values which I think could confuse the user more than just getting wrong results. Ideas are welcome.

---

WTB exceptions :) there have been several attempts, some I suspect are rather close.

I suggest a debug message and/or an assert for now.